### ISSUE TITLE
Optimizer repeatedly querying the same point

### ISSUE THREAD

**Issue by 123epsilon** [resibots/limbo#320]
I am using the experimental CBOptimizer with the ECI acquisition function, I am testing the optimization of a compressor with respect to compress ratio and the SSIM score of the reconstructed data. As a result my objective function has dim_out = 1 and nb_constraints =1 where output(0) is the compression ratio and output(1) is 1 if the threshold for ssim is met and 0 otherwise. 

I have also attempted to use the distance to target aggregator function in which case I set dim_out=2 (compress ratio, ssim) and nb_constraints=1 where the constraint is the same indicator function as before. 

After the initial random sampling step (where most of the points end up not meeting the ssim threshold, thus causing the indicator to repeatedly return 0) the CBOptimizer seems to query a few new points **but then always ends up in a state where it repeatedly queries the same point (with the same resulting output) until num_iterations is reached.**

Is there anything I can try to do to avoid this behavior? I'm not entirely sure why the optimizer insists on repeatedly querying the same point rather than explore elsewhere in the space.

Otherwise the GP has the data mean function, squaredexpard kernel, kernellfopt hyperparam optimizer, and the default nlopt nograd optimizer.

**Comment by 123epsilon** [comment]
Any ideas on this issue?

**Comment by 123epsilon** [comment]
@costashatz Hi, I think I've iterated past this point now. But it was a basic optimization loop where I just specified using the CBOptimizer and ran boptimer.optimize. I resolved the problem by not using the default NLOPT gradient-free optimizer and instead using GN_ISRES, which resolved my issue. Still not 100% sure why this issue occurred with that optimizer specifically, but I was not able to replicate that behavior with any other NLOPT gradient-free optimizer I tried. I'm actually troubleshooting a different issue right now where the boost parameter binding for the initialization function for the model does not seem to be properly binding it - I opened a separate issue on this however: #323
